src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -163,9 +163,11 @@ case class AppDefinition(
     readinessChecks.foreach { r => builder.addReadinessCheckDefinition(ReadinessCheckSerializer.toProto(r)) }
     taskKillGracePeriod.foreach { t => builder.setTaskKillGracePeriod(t.toMillis) }
 
-    val roles = Protos.ResourceRoles.newBuilder()
-    roles.addAllRole(acceptedResourceRoles.asJava)
-    builder.setAcceptedResourceRoles(roles)
+    if (acceptedResourceRoles.nonEmpty) {
+      val roles = Protos.ResourceRoles.newBuilder()
```

Comment:
If we set an empty list proto will set an empty list and `hasAcceptedResourceRoles` will be true. It broke this assertion: https://github.com/mesosphere/marathon/blob/feature/pods/src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala#L72.